geomesa-blobstore/geomesa-blobstore-api/src/main/scala/org/locationtech/geomesa/blob/api/BlobstoreServlet.scala

```diff
@@ -53,10 +70,68 @@ class BlobstoreServlet extends GeoMesaScalatraServlet with LazyLogging {
         NotFound(reason = s"Unknown ID $id")
       } else {
         contentType = "application/octet-stream"
-        response.setHeader("Content-Disposition", "attachment; filename=" + filename)
+        response.setHeader("Content-Disposition", "attachment;filename=" + filename)
 
         Ok(returnBytes)
       }
     }
   }
+
+  // TODO: Revisit configuration and persistence of configuration.
+  // https://geomesa.atlassian.net/browse/GEOMESA-958
+  // https://geomesa.atlassian.net/browse/GEOMESA-984
+  post("/ds/?") {
+    logger.debug("In ds registration method")
+
+    val dsParams = datastoreParams
+    val ds = new AccumuloDataStoreFactory().createDataStore(dsParams).asInstanceOf[AccumuloDataStore]
+
+    if (ds == null) {
+      NotFound(reason = "Could not load data store using the provided parameters.")
+    } else {
+      this.synchronized {
+        abs = new AccumuloBlobStore(ds)
+      }
+      //see https://httpstatuses.com/204
+      NoContent()
+    }
+  }
+
+  // scalatra routes bottom up, so we want the ds post to be checked first
+  post("/") {
+    try {
+      logger.debug("In file upload post method")
+      if (abs == null) {
+        NotFound(reason = "AccumuloBlobStore is not initialized.")
+      } else {
+        fileParams.get("file") match {
+          case None =>
+            halt(400, reason = "no file parameter in request")
+          case Some(file) =>
+            val otherParams = multiParams.toMap.map { case (s, p) => s -> p.head }
+            val tempFile = new File(tempDir.toString + '/' + file.getName)
+            file.write(tempFile)
+            val actRes: ActionResult = abs.put(tempFile, otherParams) match {
+              case Some(id) =>
+                Created(body = id, headers = Map("Location" -> s"${request.getRequestURL append id}"))
```

Comment:
i think we discourage the 'space' notation, e.g. should be request.getRequestURL.append(id)